cmd/performance-profile-creator/cmd/root.go

```diff
@@ -55,45 +55,90 @@ type ProfileData struct {
 // ClusterData collects the cluster wide information, each mcp points to a list of ghw node handlers
 type ClusterData map[*machineconfigv1.MachineConfigPool][]*profilecreator.GHWHandler
 
-// rootCmd represents the base command when called without any subcommands
-var rootCmd = &cobra.Command{
-	Use:   "performance-profile-creator",
-	Short: "A tool that automates creation of Performance Profiles",
-	RunE: func(cmd *cobra.Command, args []string) error {
-		mustGatherDirPath := cmd.Flag("must-gather-dir-path").Value.String()
-		cluster, err := getClusterData(mustGatherDirPath)
-		if err != nil {
-			return fmt.Errorf("failed to parse the cluster data: %v", err)
-		}
+func init() {
+	log.SetOutput(os.Stderr)
+	log.SetFormatter(&log.TextFormatter{
+		DisableTimestamp: true,
+	})
+}
 
-		info, err := strconv.ParseBool(cmd.Flag("info").Value.String())
-		if err != nil {
-			return fmt.Errorf("failed to parse the info flag: %v", err)
-		}
-		if info {
-			showClusterData(cluster)
+// NewRootCommand returns entrypoint command to interact with all other commands
+func NewRootCommand() *cobra.Command {
+	pcArgs := &ProfileCreatorArgs{
+		UserLevelNetworking: pointer.BoolPtr(false),
+	}
+
+	var requiredFlags []string = []string{
+		"reserved-cpu-count",
+		"mcp-name",
+		"rt-kernel",
+		"must-gather-dir-path",
+	}
+
+	root := &cobra.Command{
+		Use:   "performance-profile-creator",
+		Short: "A tool that automates creation of Performance Profiles",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if !cmd.Flag("must-gather-dir-path").Changed {
```

Comment:
While this makes sense from a code perspective (but looks like a sidegrade to me), from UX perspective I'm not convinced we gain much. Currently we get:
```
$ ./build/_output/bin/performance-profile-creator --info --reserved-cpu-count=2
Error: missing cluster data: please use the --must-gather-dir-path flag
Usage:
  performance-profile-creator [flags]

Flags:
[...]
```
which looks good to me because it clearly errors out and tells the user what's wrong and how to fix. Could you please elaborate a bit more the UX flow you would like to be improved?

Reply:
Well, I was hoping that we would have consistent messaging across all the required args in case of `non-info` mode.
Specifically that we should have got:

`Error: missing required flags: must-gather-dir-path, mcp-name, rt-kernel`

Reply:
ok, makes sense. FIxing.